src/server/routes/backend-wallet/send-transactions-atomic.ts

```diff
@@ -0,0 +1,151 @@
+import { Type, type Static } from "@sinclair/typebox";
+import type { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import type { Address, Hex } from "thirdweb";
+import { insertTransaction } from "../../../shared/utils/transaction/insert-transaction";
+import { AddressSchema } from "../../schemas/address";
+import {
+  requestQuerystringSchema,
+  standardResponseSchema,
+  transactionWritesResponseSchema,
+} from "../../schemas/shared-api-schemas";
+import {
+  maybeAddress,
+  walletChainParamSchema,
+  walletWithAAHeaderSchema,
+} from "../../schemas/wallet";
+import { getChainIdFromChain } from "../../utils/chain";
+import {
+  getWalletDetails,
+  isSmartBackendWallet,
+  type ParsedWalletDetails,
+  WalletDetailsError,
+} from "../../../shared/db/wallets/get-wallet-details";
+import { createCustomError } from "../../middleware/error";
+
+const requestBodySchema = Type.Object({
+  transactions: Type.Array(
+    Type.Object({
+      toAddress: Type.Optional(AddressSchema),
+      data: Type.String({
+        examples: ["0x..."],
+      }),
+      value: Type.String({
+        examples: ["10000000"],
+      }),
+    }),
+  ),
+});
+
+export async function sendTransactionsAtomicRoute(fastify: FastifyInstance) {
+  fastify.route<{
+    Params: Static<typeof walletChainParamSchema>;
+    Body: Static<typeof requestBodySchema>;
+    Reply: Static<typeof transactionWritesResponseSchema>;
+    Querystring: Static<typeof requestQuerystringSchema>;
+  }>({
+    method: "POST",
+    url: "/backend-wallet/:chain/send-transactions-atomic",
```

Comment:
There's already a `/send-transactions-batch`. IMO introducing `/send-transactions-atomic` will confuse the dev.

WDYT about reusing the same handler and branching the logic if SBW? In the docs we'd then need to clarify "Order is guaranteed for SBW only. If non-SBW, order is not guaranteed".

If you still want two separate paths, I'd prefer avoiding a new term (atomic) and using simpler language, like `/send-transaction-batch-ordered`. Or `/send-userop-batch` to make it clear it's for smart wallets.

Reply:
answered why not the same handler in a question above, Joaquim had a similar question.

I do agree the name for the other path could be improved, but I ran out of ideas.

`send-userop-batch` could be misleading since we're creating a single UserOp rather than batching multiple UserOps.

I do like keeping "atomic" in the endpoint name since it's a widely understood term beyond web3 - developers regularly encounter it in databases, filesystems, and other contexts where operations need to succeed or fail as one unit, and it's more precise than "ordered".

I was initially thinking `/send-transaction-batch-atomic` to go along with the existing `/send-transaction-batch`, but felt it would be too long.